airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -184,17 +182,23 @@ def _read_incremental(
                 stream_state = stream_instance.get_updated_state(stream_state, record_data)
                 if checkpoint_interval and record_counter % checkpoint_interval == 0:
                     yield self._checkpoint_state(stream_name, stream_state, connector_state, logger)
+                if internal_config.limit and record_counter >= internal_config.limit:
```

Comment:
so we don't need it here because of comment above

Reply:
still need it, check out updated PR